lib/pdf_utilities/datestamp_pdf.rb

```diff
@@ -7,45 +7,55 @@ module PDFUtilities
   # @see https://github.com/jkraemer/pdf-forms
   PDFTK = PdfForms.new(Settings.binaries.pdftk)
 
+  class DatestampPdfError < StandardError; end
+  class PdfMissingError < DatestampPdfError; end
+  class StampGenerationError < DatestampPdfError; end
+  class PdfStampingError < DatestampPdfError; end
+
   # add a watermark datestamp to an existing pdf
   class DatestampPdf
     # prepare to datestamp an existing pdf document
     #
-    # @param file_path [String]
-    # @param append_to_stamp [String] text to append to the stamp
+    # @param file_path [String] path to the PDF file
+    # @param append_to_stamp [String, nil] text to append to the stamp
+    # @raise [PdfMissingError] if the PDF file doesn't exist
     #
     def initialize(file_path, append_to_stamp: nil)
       @file_path = file_path
       @append_to_stamp = append_to_stamp
+
+      raise PdfMissingError, "Original PDF missing: #{file_path}" unless File.exist?(file_path)
```

Comment:
Given how often we've seen files exist and then stop existing a moment later, this seems like a great addition! 👍